File: Backend/SoftLibOrdenBO/src/pe/edu/pucp/softlib/orden/bo/OrdenBO.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.softlib.orden.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softlib.orden.dao.OrdenDAO;
import pe.edu.pucp.softlib.orden.daoImp.OrdenDAOImpl;
import pe.edu.pucp.softlib.orden.model.EstadoDeOrden;
import pe.edu.pucp.softlib.orden.model.LineaDeOrden;
import pe.edu.pucp.softlib.orden.model.Orden;
import pe.edu.pucp.softlib.usuario.model.Empleado;

/**
 * 
 * @author Joshua Haro
 */
public class OrdenBO {
    private final OrdenDAO ordenDAO;
    
    public OrdenBO(){
        this.ordenDAO = new OrdenDAOImpl();
    }
    
    public Integer insertar(ArrayList<LineaDeOrden> lineasDeOrdenes,
            EstadoDeOrden estadoDeOrden, Date fechaCreacion, Double total, 
            Integer idEmpleado){
        Empleado empleado = new Empleado();
        empleado.setIdPersona(idEmpleado);
        Orden orden = new Orden(null,lineasDeOrdenes,estadoDeOrden,fechaCreacion,
                total,empleado,true);
        return this.ordenDAO.insertar(orden);
    }
    
    public Integer modificar(Integer idOrden,ArrayList<LineaDeOrden> lineasDeOrdenes,
            EstadoDeOrden estadoDeOrden, Date fechaCreacion, Double total, 
            Integer idEmpleado, Boolean activo){
        Empleado empleado = new Empleado();
        empleado.setIdPersona(idEmpleado);
        Orden orden = new Orden(idOrden,lineasDeOrdenes,estadoDeOrden,fechaCreacion,
                total,empleado,activo);
        return this.ordenDAO.modificar(orden);
    }
    
    public Integer eliminar(Integer idOrden){
        Orden orden = new Orden();
        orden.setIdOrden(idOrden);
        return this.ordenDAO.eliminar(orden);
    }
    
    public ArrayList<Orden> listarTodos(){
        return this.ordenDAO.listarTodos();
    }
    
    public Orden obtenerPorId(Integer idOrden){
        return this.ordenDAO.obtenerPorId(idOrden);
    }
    
    public Integer existeOrden(Date fechaCreacion, Double total, Integer idEmpleado){
        Orden orden= new Orden();
        orden.setFechaCreacion(fechaCreacion);
        orden.setTotal(total);
        Empleado empleado = new Empleado();
        empleado.setIdPersona(idEmpleado);
        return this.ordenDAO.existeOrden(orden);
    }    
}

File: Frontend/SoftLibOrdenBO/OrdenBO.cs
using SoftLibBO;
using SoftLibBO.ServicioWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftLibOrdenBO
{
    public class OrdenBO : BaseBO
    {
        public int insertar(BindingList<lineaDeOrden> lineasDeOrdenes,
           estadoDeOrden estadoDeOrden, DateTime fechaCreacion, double total,
           int idEmpleado)
        {
            lineaDeOrden[] lineas = lineasDeOrdenes.ToArray();
            return this.WsOrden.orden_insertar(lineas, estadoDeOrden, fechaCreacion, total,
             idEmpleado);
        }

        public int modificar(int idOrden, BindingList<lineaDeOrden> lineasDeOrdenes,
                estadoDeOrden estadoDeOrden, DateTime fechaCreacion, double total,
                int idEmpleado, bool activo)
        {
            lineaDeOrden[] lineas = lineasDeOrdenes.ToArray();
            return this.WsOrden.orden_modificar(idOrden, lineas,
                 estadoDeOrden, fechaCreacion, total,
                 idEmpleado, activo);
        }

        public int eliminar(int idOrden)
        {
            return this.WsOrden.orden_eliminar(idOrden);
        }

        public BindingList<orden> listarTodos()
        {
            orden[] arreglo = this.WsOrden.orden_listarTodos();
            return new BindingList<orden>(arreglo);
        }

        public orden obtenerPorId(int idOrden)
        {
            return this.WsOrden.orden_obtenerPorId(idOrden);
        }

        public int existeOrden(DateTime fechaCreacion, double total, int idEmpleado)
        {
            return this.WsOrden.existeOrden(fechaCreacion, total, idEmpleado);
        }
    }
}
